### ISSUE TITLE
Deprecated methods for iOS8

### ISSUE THREAD

**Issue by rav013** [wonderpush/wonderpush-ios-sdk#3]
Please update `enabledRemoteNotificationTypes`, `registerForRemoteNotificationTypes` and `UIRemoteNotificationTypeAlert`, `UIRemoteNotificationTypeSound` are deprecated in iOS 8.0.

![Image](https://i.imgsafe.org/c6b1e9376e.png)

**Comment by ofavre** [comment]
Those calls cannot be removed without removing support for iOS 7 altogether.
The iOS version is detected to call the pre- or post- iOS 8 functions properly.

I'll use some pragma directives to silence these warnings on the appropriate function.

``` objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
- (void) methodUsingDeprecatedStuff {
    // use deprecated stuff
}
#pragma clang diagnostic pop
```

**Comment by ofavre** [comment]
Closed with 5eabaae63000c93bf46ec7239426f3a6c1d6e580.
Thanks.
